scripts/retry_make_nncp.sh

```diff
@@ -0,0 +1,15 @@
+#!/bin/bash
+
+n=0
+retries="${1:-5}"  # Number of retries with a default value of 5
+
+while true; do
+    make nncp && break
+    make nncp_cleanup
```

Comment:
i feel like this is not a stable pattern

when we delete and recreate the resource  even if its the same content its going to be a separate object at the k8s level so without an exponential backoof i.e. waiting longer each time we call make nncp i think its possible that we ould be in a loop just because we kept restarting before it could make progress.

should we increase the NNCP_TIMEOUT slightly on each retry?

Reply:
>i feel like this is not a stable pattern
It's not ideal but it's a worst case recovery, we already give the NNCP CR 240 seconds to come up and I haven't seen any jobs failing due to timeout. 

>i think its possible that we ould be in a loop just because we kept restarting before it could make progress.
I haven't seen any job fail due to NNCP CR taking over 240s to apply. This loop is limited to 5 iterations so max it would be stuck for is 20mins and this would only be if there is a real issue with the NNCP CR that needs to be fixed, aka a real bug.

>should we increase the NNCP_TIMEOUT slightly on each retry?
No I don't believe we should, I believe it's high enough already unless you have seen jobs failing to apply the NNCP in under 240 seconds?